lib/repository.py

```diff
@@ -41,6 +42,13 @@ def get_git_repository(name, remote_repo_url, parent_dir_path):
     dir, according to specified name.
     If it does not exist, clone it from the specified URL.
     """
+    # infer name if it is not provided
+    if not name:
+        match_obj = re.match(".*/([a-zA-Z0-9\-]*)\.git", remote_repo_url)
```

Comment:
imho [a-zA-Z0-9\-_]* is a safe regex, I have never seen other chars in source code repositories names